website/src/components/getFocusPath.js

```diff
@@ -34,6 +34,8 @@ export default function getFocusPath(node, pos, parser, seen = new Set()) {
     if (value && typeof value === 'object' && !seen.has(value)) {
       let childPath = getFocusPath(value, pos, parser, seen);
       if (childPath.length > 0) {
+        // if current wasn't added, add it now
+        childPath = range ? childPath : [node].concat(childPath);
```

Comment:
In OCaml AST case, not all nodes throughout the tree have location information.

This change allows to include a child in the path even if the call to `nodeToRange` above didn't return a range, but only _if we know there are deeper descendants in the path_.

This allows to provide a more accurate node selection in many cases.

Reply:
FWIW I'm actually in the process of removing this code and replace with something that uses https://github.com/fkling/astexplorer/blob/540c2311c51d451fd1d97e03b2589ebd4606ee6e/website/src/core/TreeAdapter.js#L26-L81 instead.

Reply:
Cool! it seems (from a very quick look haha) that this code would continue to cover this case, right?

Reply:
Yep!